src/v/cluster/types.h

```diff
@@ -3013,6 +3014,31 @@ constexpr auto common = partition_allocation_domain(0);
 constexpr auto consumer_offsets = partition_allocation_domain(-1);
 } // namespace partition_allocation_domains
 
+/*
+ * partition_reassignments represents the result from ListPartitionReassignments
+ * API
+ *
+ *          replicas: The current replica set
+ *   adding_replicas: The set of replicas we are currently adding
+ * removing_replicas: The set of replicas we are currently removing
+ *        error_code: A problem occurred if not kafka::error_code::none
+ *     error_message: The problem description
+ */
+struct partition_reassignments {
+    std::vector<model::node_id> replicas;
+    std::vector<model::node_id> adding_replicas;
+    std::vector<model::node_id> removing_replicas;
+    kafka::error_code error_code;
+    ss::sstring error_message;
+
+    friend bool
+    operator==(const partition_reassignments&, const partition_reassignments&)
+      = default;
+
+    friend std::ostream&
+    operator<<(std::ostream& o, const partition_reassignments& r);
+};
```

Comment:
I could just use a `ongoing_partition_reassignment` object but that does not capture errors. I'm open to suggestions.

Reply:
Generally a cluster object with a member from the `kafka::` namespace is not something we'd do, there are probably some exceptions in this file related to transactions.

Reply:
That makes sense. I'll change the error_code to be type `std::error_code` to remain consistent with design.